crates/re_sdk/src/log_sink.rs

```diff
@@ -6,47 +9,31 @@ pub trait LogSink: Send + Sync + 'static {
     fn send(&self, msg: LogMsg);
 
     /// Send all these log messages.
+    #[inline]
     fn send_all(&self, messages: Vec<LogMsg>) {
         for msg in messages {
             self.send(msg);
         }
     }
 
     /// Drain all buffered [`LogMsg`]es and return them.
+    ///
+    /// Only applies to sinks that maintain a backlog.
+    #[inline]
     fn drain_backlog(&self) -> Vec<LogMsg> {
         vec![]
     }
 
-    /// Wait until all logged data have been sent to the remove server (if any).
-    fn flush(&self) {}
-
-    /// If the TCP session is disconnected, allow it to quit early and drop unsent messages.
-    fn drop_msgs_if_disconnected(&self) {}
-
-    /// Returns `false` if this sink just discards all messages.
-    fn is_enabled(&self) -> bool {
-        true
-    }
-}
-
-// ----------------------------------------------------------------------------
-
-struct DisabledSink;
-
-impl LogSink for DisabledSink {
-    fn send(&self, _msg: LogMsg) {
-        // It's intended that the logging SDK should drop messages earlier than this if logging is disabled.
-        re_log::debug_once!("Logging is disabled, dropping message(s).");
-    }
-
-    fn is_enabled(&self) -> bool {
-        false
-    }
-}
+    /// Blocks until all pending data in the sink's send buffers has been fully flushed.
+    ///
+    /// See also [`LogSink::drop_if_disconnected`].
+    #[inline]
+    fn flush_blocking(&self) {}
```

Comment:
I'm inclined to leave out the default impl here and force sinks to be explicit about their flush behavior rather than just silently failing to flush.